qutip/solver/brmesolve.py

```diff
@@ -0,0 +1,212 @@
+
+    tlist : array_like
+        List of times for evaluating evolution
+
+    a_ops : list of (a_op, spectra)
+        Nested list of system operators that couple to the environment,
+        and the corresponding bath spectra.
+
+        a_op : :class:`qutip.Qobj`, :class:`qutip.QobjEvo`
+            The operator coupling to the environment. Must be hermitian.
+
+        spectra : :class:`Coefficient`, str, func
+            The corresponding bath spectral responce.
+            Can be a `Coefficient` using an 'w' args, a function of the
+            frequence or a string. Coefficient build from a numpy array are
+            understood as a function of ``w`` instead of ``t``. Function are
+            expected to be of the signature ``f(w)`` or ``f(t, w, **args)``.
+
+            The spectra function can depend on ``t`` if the corresponding
+            ``a_op`` is a :cls:`QobjEvo`.
+
+        Example:
+            a_ops = [
+                (a+a.dag(), ('w>0', args={"w": 0})),
+                (QobjEvo(a+a.dag()), 'w > exp(-t)'),
+                (QobjEvo([b+b.dag(), f(t)]), g(w)),
+                (c+c.dag(), SpectraCoefficient(coefficient(array, tlist=ws))),
+            ]
+
+    .. note:
+        ``Cubic_Spline`` have been replaced by numpy array :cls:`Coefficient`.
+        Whether the ``a_ops`` is time dependent is deceided by the type of the
+        operator: :cls:`Qobj` vs :cls:`QobjEvo` instead of the type of the
+        spectra.
+
+    e_ops : list of :class:`Qobj` / callback function
+        Single operator or list of operators for which to evaluate
+        expectation values or callable or list of callable.
+        Callable signature must be, `f(t: float, state: Qobj)`.
+        See :func:`expect` for more detail of operator expectation
+
+    c_ops : list of (:class:`QobjEvo`, :class:`QobjEvo` compatible format)
+        List of collapse operators.
+
+    args : dict
+        Dictionary of parameters for time-dependent Hamiltonians and
+        collapse operators. The key ``w`` is reserved for the spectra function.
+
+    sec_cutoff : float {0.1}
+        Cutoff for secular approximation. Use ``-1`` if secular approximation
+        is not used when evaluating bath-coupling terms.
+
+    options : :class:`qutip.solver.SolverOptions`
+        Options for the solver.
+
+    Returns
+    -------
+    result: :class:`qutip.solver.Result`
+
+        An instance of the class :class:`qutip.solver.Result`, which contains
+        either an array of expectation values, for operators given in e_ops,
+        or a list of states for the times specified by `tlist`.
+
+    .. note:
+        operator_data_type
+    """
+    H = QobjEvo(H, args=args, tlist=tlist)
+
+    c_ops = c_ops if c_ops is not None else []
+    if not isinstance(c_ops, (list, tuple)):
+        c_ops = [c_ops]
+    c_ops = [QobjEvo(c_op, args=args, tlist=tlist) for c_op in c_ops]
+
+    new_a_ops = []
+    for (a_op, spectra) in a_ops:
```

Comment:
We should check that `a_op` is either `Qobj` or `QobjEvo` here, or try convert it to a QobjEvo perhaps.